troposphere/static/js/components/modals/instance/launch/steps/NameIdentityVersionStep.react.js

```diff
@@ -1,267 +1,263 @@
-                        <div className={"progress-bar " + barTypeClass} style={currentlyUsedStyle}></div>
-                        <div className={"progress-bar " + barTypeClass} style={projectedUsedStyle}></div>
-                    </div>
-                    <div>{message}</div>
+        return (
+            <div className="allocation-consumption-bar">
+                <div className="progress">
+                    <div className="value">{Math.round(currentlyUsedPercent + projectedPercent) + "%"}</div>
+                    <div className={"progress-bar " + barTypeClass} style={currentlyUsedStyle}></div>
+                    <div className={"progress-bar " + barTypeClass} style={projectedUsedStyle}></div>
                 </div>
-            );
-        },
+                <div>{message}</div>
+            </div>
+        );
+    },
 
-        focusInput: function() {
-          var nameField = this.refs.nameField;
-          if(nameField) {
-            nameField.getDOMNode().focus();
-          }
-        },
-        renderBody: function () {
-            var image = this.props.image,
-                identityID,
-                identities = stores.IdentityStore.getAll(),
-                providers = stores.ProviderStore.getAll(),
-                versions = stores.ImageStore.getVersions(image.id);
+    focusInput: function() {
+        var nameField = this.refs.nameField;
+        if(nameField) {
+        nameField.getDOMNode().focus();
+        }
+    },
+    renderBody: function () {
+        var image = this.props.image,
+            identityID,
+            identities = stores.IdentityStore.getAll(),
+            providers = stores.ProviderStore.getAll(),
+            versions = stores.ImageStore.getVersions(image.id);
+        
 
-            if (!providers || !identities || !versions) return <div className="loading"></div>;
+        if (!providers || !identities || !versions) return <div className="loading"></div>;
 
-            // don't show duplicate images
-            versions = this.cleanVersions(versions);
-            //Keep things in order
-            versions = versions.sort();
-            identities = identities.sort();
-            if (!this.state.version) {
-                this.state.version = versions.last();
-            }
-            if (!this.state.identity) {
-                //NOTE: DO NOT set an identity if no version can be found.
-                if(this.state.version) {
-                    this.state.identity = identities.first();
-                    identityID = this.state.identity.id;
-                } else {
-                    identityID = -1;
-                }
+        // don't show duplicate images
+        versions = this.cleanVersions(versions);
+        //Keep things in order
+        versions = filterEndDate( versions.sort() );
+        identities = filterEndDate( identities.sort() );
```

Comment:
Sort does work on the returned collection of filterEndDate but working inside out seems easer to read.